src/sentry/api/endpoints/organization_group_index.py

```diff
@@ -183,7 +184,45 @@ def get(self, request, organization):
 
         results = list(cursor_result)
 
-        context = serialize(results, request.user, serializer())
+        skip_snuba_fields = {
+            "query",
+            "status",
+            "bookmarked_by",
+            "assigned_to",
+            "unassigned",
+            "subscribed_by",
+            "active_at",
+            "first_release",
+            "first_seen",
+            "message",  # not 100% on this one
+        }
+        snuba_filters = []
+        if query_kwargs["search_filters"] is not None:
+            snuba_filters = [
+                convert_search_filter_to_snuba_query(search_filter)
+                for search_filter in query_kwargs["search_filters"]
+                if search_filter.key.name not in skip_snuba_fields
+            ]
+
+        lifetime_stats = serialize(results, request.user, serializer())
+
+        if features.has("organizations:dynamic-issue-counts", organization, actor=request.user):
+            context = serialize(results, request.user, serializer(start=start, end=end))
```

Comment:
I don't think we should be serializing all of the groups twice, since we're throwing away most of the serialization on the second pass, and we'll make a bunch of postgres queries doing so.

We could potentially just include these extra fields as part of `StreamGroupSerializerSnuba`, it is only used in `OrganizationGroupIndexEndpoint` anyway. This might be irrelevant if we can fetch these stats in a single query as @fpacifici suggested.

Reply:
Since we are on the conversation of performance, it might be a good idea to track performance of the page for sentry before an after this launch to make sure there isn't a significant hit here. @iProgramStuff & @davidenwang are working on tracking these efforts anyway, this might be a good opportunity to make sure their efforts are useful to the team :-)